frontend/packages/knative-plugin/src/components/overview/__tests__/ConfigurationsOverviewListItem.spec.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { ResourceLink } from '@console/internal/components/utils';
+import { ConfigurationModel } from '@console/knative-plugin';
+import ConfigurationsOverviewListItem from '../ConfigurationsOverviewListItem';
+import { mockConfigurationsData } from '../__mocks__/overview-knative-mock';
+
+describe('ConfigurationsOverviewListItem', () => {
+  it('should component exists', () => {
+    const wrapper = shallow(
+      <ConfigurationsOverviewListItem configuration={mockConfigurationsData[0]} />,
+    );
+    expect(wrapper.exists()).toBeTruthy();
+  });
+
+  it('should list the Configuration', () => {
+    const wrapper = shallow(
+      <ConfigurationsOverviewListItem configuration={mockConfigurationsData[0]} />,
+    );
+    expect(wrapper.find('li')).toHaveLength(1);
+    expect(
+      wrapper
+        .find('li')
+        .at(0)
+        .props().className,
+    ).toEqual('list-group-item');
+  });
+
+  it('should have ResourceLink with proper kind', () => {
+    const wrapper = shallow(
+      <ConfigurationsOverviewListItem configuration={mockConfigurationsData[0]} />,
+    );
+    expect(wrapper.find(ResourceLink)).toHaveLength(1);
+    expect(
+      wrapper
+        .find(ResourceLink)
+        .at(0)
+        .props().kind,
+    ).toEqual(referenceForModel(ConfigurationModel));
+  });
+
+  it('should display latestCreatedRevisionName and latestReadyRevisionName', () => {
+    const wrapper = shallow(
+      <ConfigurationsOverviewListItem configuration={mockConfigurationsData[0]} />,
+    );
+    expect(wrapper.find('span')).toHaveLength(4);
+    expect(
+      wrapper
+        .find('span')
+        .at(1)
+        .props().children,
+    ).toEqual(mockConfigurationsData[0].status.latestCreatedRevisionName);
```

Comment:
I'd like to avoid finding a random tag and then assert its contents when really we just want assert that there is some text in the entirety of the rendered content.

@rohitkrai03 @andrewballantyne how much asserting of text content do you find useful?

If we do agree to do these assertions, we should just look for the text in the rendered content to avoid coupling it to the structure. Something like this: `expect(wrapper.text().includes(mockConfigurationsData[0].status.latestCreatedRevisionName)).toBe(true);`

Or something else...

Reply:
Testing static, immutable, always-there, text? I don't see much value. If we test at this level, our tests probably should cover static markup and other things that "the user sees".

I think if we want to test static markup/text, it needs to be a conditional markup/text. Like "I lack the information, present an error to the user" kinda thing.

- my $0.02

Reply:
This was along my thinking as well. Asserting the text if there is a condition in which the text changes.